File: code_challenge_solution/c#/194 Temperature converter.cs
using System;

class Program
{
    static void Main(string[] args)
    {
        Console.WriteLine("Temperature Converter");
        Console.WriteLine("----------------------");

        Console.WriteLine("1. Celsius to Fahrenheit");
        Console.WriteLine("2. Fahrenheit to Celsius");
        Console.WriteLine("3. Celsius to Kelvin");
        Console.WriteLine("4. Kelvin to Celsius");
        Console.WriteLine("5. Fahrenheit to Kelvin");
        Console.WriteLine("6. Kelvin to Fahrenheit");

        Console.Write("Enter your choice: ");
        int choice = Convert.ToInt32(Console.ReadLine());

        double temperature, convertedTemperature;

        switch (choice)
        {
            case 1:
                Console.Write("Enter temperature in Celsius: ");
                temperature = Convert.ToDouble(Console.ReadLine());
                convertedTemperature = CelsiusToFahrenheit(temperature);
                Console.WriteLine("Temperature in Fahrenheit: " + convertedTemperature);
                break;
            case 2:
                Console.Write("Enter temperature in Fahrenheit: ");
                temperature = Convert.ToDouble(Console.ReadLine());
                convertedTemperature = FahrenheitToCelsius(temperature);
                Console.WriteLine("Temperature in Celsius: " + convertedTemperature);
                break;
            case 3:
                Console.Write("Enter temperature in Celsius: ");
                temperature = Convert.ToDouble(Console.ReadLine());
                convertedTemperature = CelsiusToKelvin(temperature);
                Console.WriteLine("Temperature in Kelvin: " + convertedTemperature);
                break;
            case 4:
                Console.Write("Enter temperature in Kelvin: ");
                temperature = Convert.ToDouble(Console.ReadLine());
                convertedTemperature = KelvinToCelsius(temperature);
                Console.WriteLine("Temperature in Celsius: " + convertedTemperature);
                break;
            case 5:
                Console.Write("Enter temperature in Fahrenheit: ");
                temperature = Convert.ToDouble(Console.ReadLine());
                convertedTemperature = FahrenheitToKelvin(temperature);
                Console.WriteLine("Temperature in Kelvin: " + convertedTemperature);
                break;
            case 6:
                Console.Write("Enter temperature in Kelvin: ");
                temperature = Convert.ToDouble(Console.ReadLine());
                convertedTemperature = KelvinToFahrenheit(temperature);
                Console.WriteLine("Temperature in Fahrenheit: " + convertedTemperature);
                break;
            default:
                Console.WriteLine("Invalid choice");
                break;
        }
    }

    static double CelsiusToFahrenheit(double celsius)
    {
        return (celsius * 9 / 5) + 32;
    }

    static double FahrenheitToCelsius(double fahrenheit)
    {
        return (fahrenheit - 32) * 5 / 9;
    }

    static double CelsiusToKelvin(double celsius)
    {
        return celsius + 273.15;
    }

    static double KelvinToCelsius(double kelvin)
    {
        return kelvin - 273.15;
    }

    static double FahrenheitToKelvin(double fahrenheit)
    {
        double celsius = FahrenheitToCelsius(fahrenheit);
        return CelsiusToKelvin(celsius);
    }

    static double KelvinToFahrenheit(double kelvin)
    {
        double celsius = KelvinToCelsius(kelvin);
        return CelsiusToFahrenheit(celsius);
    }
}

File: code_challenge_solution/c++/194 Temperature converter.cpp
```cpp
#include <iostream>

double celsiusToFahrenheit(double celsius) {
    return (celsius * 9 / 5) + 32;
}

double fahrenheitToCelsius(double fahrenheit) {
    return (fahrenheit - 32) * 5 / 9;
}

double celsiusToKelvin(double celsius) {
    return celsius + 273.15;
}

double kelvinToCelsius(double kelvin) {
    return kelvin - 273.15;
}

double fahrenheitToKelvin(double fahrenheit) {
    return (fahrenheit + 459.67) * 5 / 9;
}

double kelvinToFahrenheit(double kelvin) {
    return (kelvin * 9 / 5) - 459.67;
}

int main() {
    double temperature;
    int choice;

    std::cout << "Temperature Converter" << std::endl;

    std::cout << "Enter temperature: ";
    std::cin >> temperature;

    std::cout << "Available conversions:" << std::endl;
    std::cout << "1. Celsius to Fahrenheit" << std::endl;
    std::cout << "2. Fahrenheit to Celsius" << std::endl;
    std::cout << "3. Celsius to Kelvin" << std::endl;
    std::cout << "4. Kelvin to Celsius" << std::endl;
    std::cout << "5. Fahrenheit to Kelvin" << std::endl;
    std::cout << "6. Kelvin to Fahrenheit" << std::endl;

    std::cout << "Enter your choice: ";
    std::cin >> choice;

    double convertedTemperature;
    switch (choice) {
        case 1:
            convertedTemperature = celsiusToFahrenheit(temperature);
            std::cout << temperature << " degree Celsius is equal to " << convertedTemperature << " degree Fahrenheit" << std::endl;
            break;
        case 2:
            convertedTemperature = fahrenheitToCelsius(temperature);
            std::cout << temperature << " degree Fahrenheit is equal to " << convertedTemperature << " degree Celsius" << std::endl;
            break;
        case 3:
            convertedTemperature = celsiusToKelvin(temperature);
            std::cout << temperature << " degree Celsius is equal to " << convertedTemperature << " Kelvin" << std::endl;
            break;
        case 4:
            convertedTemperature = kelvinToCelsius(temperature);
            std::cout << temperature << " Kelvin is equal to " << convertedTemperature << " degree Celsius" << std::endl;
            break;
        case 5:
            convertedTemperature = fahrenheitToKelvin(temperature);
            std::cout << temperature << " degree Fahrenheit is equal to " << convertedTemperature << " Kelvin" << std::endl;
            break;
        case 6:
            convertedTemperature = kelvinToFahrenheit(temperature);
            std::cout << temperature << " Kelvin is equal to " << convertedTemperature << " degree Fahrenheit" << std::endl;
            break;
        default:
            std::cout << "Invalid choice" << std::endl;
            break;
    }

    return 0;
}
```